content/docs/ops/runbook/rotating-cloudfoundry-diego.md

```diff
@@ -0,0 +1,183 @@
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.server_cert`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.server_key`
+    - Deploy this only after a successful deployment of `consul.ca_cert`.
+- `consul.encrypt_keys`
+    - Deploy the new encrypt key before the old one for the first deployment.
+      And remove the old encrypt key for the last deployment.
+
+Make sure that Diego deploys after each rotation step for the certificate
+updates from _Security Configuration for Consul_ in order to prevent downtime.
+
+#### Updating certificates in secrets
+
+Replace all the certificates corresponding to the filename and property matching
+in the secrets file, e.g. `certs_dir/cloud-controller.crt` maps to
+`properties.cc.public_cert`. Be sure to include the `-----BEGIN
+CERTIFICATE-----` and `-----END CERTIFICATE----- ` and indent the value under
+the YAML stanza to ensure valid YAML syntax.
+
+#### Remaining secrets
+
+The remaining secrets in this file besides certificates can all be rotated at
+once. Do this ahead of the first deployment so that subsequent deployments only
+need to follow the _Security Configuration for Consul_.
+
+### External secrets (external.yml)
+
+External secrets fall into a spectrum categories which ranges from _rotatable
+with coordination_ with other cloud.gov deployments on one end and _non-rotatable
+without data loss_ on the other.
+
+#### Properties used in other deployments
+
+1. `uaa.clients.*.secrets` properties need to be coordinated across various Bosh
+   deployments and cloud.gov apps or service brokers.
+1. `login.smtp.*` properties for Mandrill.
+1. `uaa.scim.users.broker-deployer` secret needs to be coordinated across our
+   Service Broker deployments for cloud.gov.
+
+#### New Relic external accounts
+
+1. `collector.newrelic_insights.*` properties NewRelic license keys and
+   configuration.
+1. `collector.newrelic_plugin.*` properties NewRelic license keys and
+   configuration.
+1. `uaa.newrelic.*` properties NewRelic license keys and configuration.
+1. `newrelic.*` properties NewRelic license keys and configuration.
+
+#### Non-rotatable properties
+1. `cc.db_encryption_key` **cannot be rotated** once set on a running CF deployment.
+    - There is currently no way to rotate this encryption key for the Cloud
+      Controller database.
+1. `login.providers.*` metadata along with `login.saml.*` properties for
+   certificates and keys.
+    - The metadata comes from our external SAML providers. [These properties
+      cannot _currently_ be rotated without downtime](https://www.pivotaltracker.com/n/projects/997278/stories/142082611).
+
+#### Rotating external UAA client secrets
+
+You will need to coordinate these changes with secrets and passphrase changes
+for various other deployments. You can [reference the list of external
+clients](https://github.com/18F/cg-deploy-cf/blob/master/cf-secrets-example.external.yml#L60-L87)
```

Comment:
I wouldn't hardcode the lines for this section of YML. maybe put comments in the YML for which subset of clients you are talking about.